internal/dag/httpproxy_processor.go

```diff
@@ -250,8 +250,17 @@ func (p *HTTPProxyProcessor) computeHTTPProxy(proxy *contour_api_v1.HTTPProxy) {
 
 			svhost := p.dag.EnsureSecureVirtualHost(listener.Name, host)
 			svhost.Secret = sec
+
 			// default to a minimum TLS version of 1.2 if it's not specified
-			svhost.MinTLSVersion = annotation.MinTLSVersion(tls.MinimumProtocolVersion, "1.2")
+			svhost.MinTLSVersion = annotation.TLSVersion(tls.MinimumProtocolVersion, "1.2")
+
+			// default to a maximum TLS version of 1.3 if it's not specified
+			svhost.MaxTLSVersion = annotation.TLSVersion(tls.MaximumProtocolVersion, "1.3")
+			if svhost.MaxTLSVersion < svhost.MinTLSVersion {
+				validCond.AddError(contour_api_v1.ConditionTypeTLSError, "TLSConfigNotValid",
+					"Spec.Virtualhost.TLS the minimum protocol version is greater than the maximum protocol version")
+				return
+			}
```

Comment:
Let's add a test case to either status_test.go or to a feature test that covers this.